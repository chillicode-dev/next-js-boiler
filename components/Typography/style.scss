.Typography {
  color: currentColor;
}

.heading1 {
  @include h1();
}

.heading2 {
  @include h2();
}

.heading3 {
  @include h3();
}

.heading4,
.heading5,
.heading6 {
  @include h4();
}

.paragraph {
  @include paragraph();
}

.heading1.hasMargin,
.heading2.hasMargin,
.heading3.hasMargin,
.heading4.hasMargin,
.heading5.hasMargin,
.heading6.hasMargin,
.paragraph.hasMargin {
  @include addTextMargin();
}

.uppercase {
  text-transform: uppercase;
}

.zeroLineHeight {
  line-height: 1;
}

.secondary {
  color: $secondaryTextColor;
}
