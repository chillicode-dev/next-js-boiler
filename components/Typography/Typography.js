// Vendor
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import parseHtml from 'react-html-parser';
// Styles
import style from './style.scss';

const getTagNameFromVariant = variant => {
  switch (variant) {
    case 'heading1':
      return 'h1';
    case 'heading2':
      return 'h2';
    case 'heading3':
      return 'h3';
    case 'heading4':
      return 'h4';
    case 'heading5':
      return 'h5';
    case 'heading6':
      return 'h6';
    case 'paragraph':
      return 'p';
    default:
      return 'div';
  }
};

const Typography = ({
  className,
  children,
  variant,
  tagName,
  hasHtml,
  hasMargin,
  uppercase,
  secondary,
  zeroLineHeight,
  ...props
}) => {
  const TagName = tagName ? tagName : getTagNameFromVariant(variant);
  const classes = cn({
    [style.Typography]: true,
    [style[variant]]: variant,
    [style.uppercase]: uppercase,
    [style.hasMargin]: hasMargin,
    [style.secondary]: secondary,
    [style.zeroLineHeight]: zeroLineHeight,
    [className]: className,
  });

  return (
    <TagName className={classes} {...props}>
      {hasHtml ? parseHtml(children) : children}
    </TagName>
  );
};

Typography.defaultProps = {
  className: '',
  tagName: null,
  secondary: false,
  hasHtml: false,
  hasMargin: false,
  uppercase: false,
  zeroLineHeight: false,
};

Typography.propTypes = {
  children: PropTypes.node.isRequired,
  variant: PropTypes.oneOf(['heading1', 'heading2', 'heading3', 'heading4', 'heading5', 'heading6', 'paragraph'])
    .isRequired,
  className: PropTypes.string,
  tagName: PropTypes.oneOfType([PropTypes.string, PropTypes.elementType]),
  hasHtml: PropTypes.bool,
  hasMargin: PropTypes.bool,
  uppercase: PropTypes.bool,
  secondary: PropTypes.bool,
  zeroLineHeight: PropTypes.bool,
};

export default Typography;
