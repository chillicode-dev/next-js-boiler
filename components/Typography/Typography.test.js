import React from 'react';
import renderer from 'react-test-renderer';
import Typography from './Typography';

it('Renders Typography correctly', () => {
  const elements = (
    <div>
      <Typography variant="heading1" hasMargin>
        Text
      </Typography>
      <Typography variant="heading2" hasMargin>
        Text
      </Typography>
      <Typography variant="heading3" hasMargin>
        Text
      </Typography>
      <Typography variant="heading4" hasMargin>
        Text
      </Typography>
      <Typography variant="heading5" hasMargin>
        Text
      </Typography>
      <Typography variant="heading6" hasMargin>
        Text
      </Typography>
      <Typography variant="heading6" hasMargin secondary>
        Text
      </Typography>
      <Typography variant="paragraph" hasMargin>
        Text
      </Typography>
      <Typography variant="paragraph" zeroLineHeight>
        Text
      </Typography>
    </div>
  );
  const tree = renderer.create(elements).toJSON();
  expect(tree).toMatchSnapshot();
});
