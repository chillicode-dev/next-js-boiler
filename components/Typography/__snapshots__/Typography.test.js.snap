// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Typography correctly 1`] = `
<div>
  <h1
    className="Typography heading1 hasMargin"
  >
    Text
  </h1>
  <h2
    className="Typography heading2 hasMargin"
  >
    Text
  </h2>
  <h3
    className="Typography heading3 hasMargin"
  >
    Text
  </h3>
  <h4
    className="Typography heading4 hasMargin"
  >
    Text
  </h4>
  <h5
    className="Typography heading5 hasMargin"
  >
    Text
  </h5>
  <h6
    className="Typography heading6 hasMargin"
  >
    Text
  </h6>
  <h6
    className="Typography heading6 hasMargin secondary"
  >
    Text
  </h6>
  <p
    className="Typography paragraph hasMargin"
  >
    Text
  </p>
  <p
    className="Typography paragraph zeroLineHeight"
  >
    Text
  </p>
</div>
`;
